### DISCUSSION TITLE
Integrating device location services

### DISCUSSION THREAD

**Discussion by adamjehu** [Budibase/budibase#2982] [Ideas]
My suggestion is for enabling geolocation as a form element which can be saved to a database. Once the device location services can be accessed, perhaps the latitude and longitude can be bound to two fields which can then be submitted to their respective columns on a Save action. If something like this can be made possible, then the saved coordinates can be used in GIS or a mapping API. Would it be possible to have a feature like this included in upcoming releases?

**Comment by suparagroup** [comment]
is there a current workaround to this? to get the lat and lon

**Comment by melohagan** [comment]
Hi @suparagroup 

It's not ideal, but the workaround would be to use an Embed component: https://docs.budibase.com/docs/embed

For example on custom button click:

**Embed 1**
```
<button style="width: 100px; height: 100px;; background-color: red;" onclick = "javascript: {
function getLocation() {
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(showPosition);
  }
}

function showPosition(position) {
  document.getElementById('custom_div').innerHTML = 'Latitude: ' + position.coords.latitude + 
  '<br>Longitude: ' + position.coords.longitude;
}
getLocation();
}
">
```

**Embed 2**
`<div id="custom_div">Change Me</div>`

![location_embed](https://user-images.githubusercontent.com/101575380/170053211-5f8da4d9-735c-445a-9c3e-614ce7790f8c.gif)

**Reply by daniel-bi** [reply]
@melohagan your solution is excellent, but I'm a donkey! rs
I need that the longitude and latitude date wil be input in collumns latitude and longitude of table, in Budibase and I don´t know to do.
My difficult is export the date from embed to New Text Field component or directly to collumns. Can you help me?

**Reply by melohagan** [reply]
Since this answer was given, the best approach to this would be to build a custom component. Specifically a [custom field component](https://docs.budibase.com/docs/custom-component#example-custom-form-field-component)

You can also have a look on the [plugins](https://github.com/Budibase/plugins) page to see if what you're looking for is already available.

**Comment by melohagan** [comment]
@daniel-bi 

Looks like @andz-bb has this one covered! 😀 

https://github.com/andz-bb/budibase-component-geolocation

See this page for seeing how to import the plugin: https://docs.budibase.com/docs/custom-plugin
